include/wil/windowing.h

```diff
@@ -0,0 +1,164 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+#ifndef __WIL_WINDOWING_INCLUDED
+#define __WIL_WINDOWING_INCLUDED
+
+#include <WinUser.h>
+#include <exception>
+
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+namespace wil
+{
+namespace details
+{
+    template <typename T>
+    struct always_false : wistd::false_type
+    {
+    };
+
+    template <typename TEnumApi, typename TCallback>
+    void CallCallbackNoThrow(TEnumApi&& enumApi, TCallback&& callback) noexcept

```

Comment:
These functions are more specific than a generic "invoke callback" - suggest renaming to something like `EnumerateWindows`, `DoEnumWindows`, `PerformWindowEnumeration`, etc.

Reply:
thanks, addressed